src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -277,6 +290,45 @@ private IList<ContentItem> FindItemsImplementation(PatternSet definition, IEnume
             return itemsList;
         }
 
+        private string GetRelatedFileExtensionProperty(string assemblyPath, IEnumerable<Asset> assets)
```

Comment:
Is it possible for this method to use file system APIs like `Path.GetFileNameWithoutExtension()` or `Path.ChangeExtension()`?  The custom logic here worries me and so I'd rather see us call methods that are well tested and proven.

Reply:
The trick is there might be extensions like `.dll.config`, `.exe.config` (according to the [spec](https://github.com/NuGet/Home/blob/dev/proposed/2021/NuGetConsumePDBInPackageReference.md#2-will-nuget-list-not-only-pdb-and-xml-but-also-allthe-extensions-in-allowedreferencerelatedfileextensionsif-there-is-any-pdb-xml-pri-dllconfig-execonfig), or [AllowedReferenceRelatedFileExtensions](https://github.com/dotnet/msbuild/blame/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L596-L601))
But the [Path.GetFileNameWithoutExtension()](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?view=net-6.0) only minus the **last period** (.) and all characters following it.